File: backend/apps/accounts/roles.py
class Roles:
    CUSTOMER = "customer"
    RECEPTIONIST = "receptionist"
    STAFF = "staff"
    MANAGER = "manager"

    CHOICES = (
        (CUSTOMER, "Customer"),
        (RECEPTIONIST, "Receptionist"),
        (STAFF, "Staff"),
        (MANAGER, "Manager"),
    )


def has_role(user, *roles):
    return bool(getattr(user, "is_authenticated", False) and getattr(user, "role", None) in roles)


def is_customer(user):
    return has_role(user, Roles.CUSTOMER)


def is_receptionist(user):
    return has_role(user, Roles.RECEPTIONIST)


def is_staff_member(user):
    return has_role(user, Roles.STAFF)


def is_manager(user):
    return has_role(user, Roles.MANAGER)

File: fonend/frontend/src/constants/roles.ts
export const ROLES = {
  CUSTOMER: "customer",
  RECEPTIONIST: "receptionist",
  STAFF: "staff",
  MANAGER: "manager",
} as const;

export type Role = (typeof ROLES)[keyof typeof ROLES];

export const ROLE_LABELS: Record<Role, string> = {
  customer: "Customer",
  receptionist: "Receptionist",
  staff: "Staff",
  manager: "Manager",
};
